oci/multi_region.go

```diff
@@ -182,12 +180,17 @@ func listAllCompartments(ctx context.Context, d *plugin.QueryData) ([]identity.C
 		},
 	}
 
-	endpointRegion := GetConfig(d.Connection).Regions[0]
+	// fetch the first region provided in config if available
+	var endpointRegion string
+	if GetConfig(d.Connection).Regions != nil {
+		endpointRegion = GetConfig(d.Connection).Regions[0]
+	}
+
 	pagesLeft := true
 	for pagesLeft {
 		response, err := session.IdentityClient.ListCompartments(ctx, request)
 		if err != nil {
-			if strings.Contains(err.Error(), "no such host") {
+			if strings.Contains(err.Error(), "no such host") && endpointRegion != "" {
```

Comment:
@bigdatasourav why are we adding the `endpointRegion != ""` check? Is it necessary?